Assignments/Program.cs

```diff
@@ -1 +1,78 @@
-﻿Console.WriteLine("Hello, World!");
+﻿// ---------------------------------------------------------------------------------------
+// T11 - Digital Root
+// ---------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Spark;
+
+public class T10 {
+   public static void Main () {
+      Console.OutputEncoding = System.Text.Encoding.UTF8;
+      Console.WriteLine ("\x1B[4m" + "Digital Root:" + "\x1B[0m" +
+         "\nThe digital root or digital sum of a non-negative integer is the " +
+         "single-digit value obtained by an iterative process of summing digits.");
+      int[] testCases = new int[] { 0, 1, 9, 21, 321, 4321, 54321, 654321, 7654321, 87654321, 987654321 };
+      for (int i = 0; i < testCases.Length; i++) {
+         Console.Write ($"{i}. dgrt({testCases[i]}) = ");
+         Console.WriteLine ($"{DigitalRootOf (testCases[i])}", Console.ForegroundColor = ConsoleColor.Cyan);
+         Console.ResetColor ();
+      }
+      Console.Write ("\nNow, lets try! Please enter a number: ");
+      if (long.TryParse (Console.ReadLine (), out long input) && input > 0) {
+         Console.Write ($"dgrt({input}) = ");
+         Console.WriteLine ($"{DigitalRootOf (input)}", Console.ForegroundColor = ConsoleColor.Cyan);
+         Console.ResetColor ();
+         if (input > 9) {
+            Console.Write ("Show the steps (y/n)? ");
+            if (char.ToLower (Console.ReadKey ().KeyChar) == 'y') {
+               Console.WriteLine ("\n\n" + "\x1B[4m" + "Steps:" + "\x1B[0m");
+               ShowSteps (input);
+            }
+         }
+      } 
+      else Console.WriteLine ("Incorrect Input", Console.ForegroundColor = ConsoleColor.Red);
+      Console.ResetColor ();
+   }
+
+   /// <summary>Returns the digital root of a input number</summary>
+   public static long DigitalRootOf (long input) {
+      if (input == 0) return 0;
+      else if (input % 9 == 0 && input > 0) return 9;

```

Comment:
![image](https://github.com/vikrampg1710-tmm/Training/assets/86363064/2a7eb471-a32c-48d8-86e1-1a12d97d47da)

Use the second part of the formula as it will avoid having to mod by 9 twice